lib/poller.js

```diff
@@ -1,80 +1,105 @@
+const fetch = require('isomorphic-fetch')
 const HttpErrors = require('http-errors')
 const moment = require('moment')
-const humanizeDuration = require('humanize-duration')
 
-exports.do = (uuid, accessToken, apiUrl, pollStep = 1000, timeout = 30000) => {
-  let pollIntervalID = null
-  let timeoutID = null
+const { timer } = require('./util')
 
-  const clearActions = () => {
-    clearTimeout(timeoutID)
-    clearInterval(pollIntervalID)
-  }
+/**
+ * Throws timeout error.
+ * @param {Number} timeout Timeout [ms].
+ * @param {String} uuid Analysis UUID to use in the error message.
+ */
+function failOnTimeout (timeout, uuid) {
+  const t = moment.duration(timeout).as('seconds')
+  /* eslint-disable no-throw-literal */
+  throw (
+    `User-specified or default time out reached after ${t} seconds.\n` +
+    'Analysis continues on server and may have completed; so run again?\n' +
+    `For status reference, UUID is ${uuid}\n`
+  )
+  /* eslint-enable no-throw-literal */
+}
 
-  const timeoutPromise = new Promise((resolve, reject) => {
-    timeoutID = setTimeout(() => {
-      clearActions()
-      const duration = humanizeDuration(moment.duration(timeout)
-        .asMilliseconds())
-      // eslint-disable-next-line prefer-promise-reject-errors
-      reject(`User-specified or default time out reached after ${duration}.\n` +
-             'Analysis continues on server and may have completed; so run again?\n' +
-             `For status reference, UUID is ${uuid}\n`)
-    }, timeout)
-  })
+/**
+ * Handles error responses, throwing errors, if necessary.
+ * @param {Object} response HTTP response.
+ */
+async function handleErrors (response, accessToken) {
+  const { status } = response
+  if (status < 200 || status > 299) {
+    let msg
+    switch (status) {
+      case 404:
+        msg = `Unauthorized analysis request, access token: ${accessToken}`
```

Comment:
This message seems wrong for 404

Reply:
@rocky asked me to not modify existing error message. Otherwise, entire switch block can be removed, and the text error message just be fetched from API response as `(await response.json()).error`, which is now the default case of the switch.

Reply:
Ok. I can take care of this later when I try it out. Basically though I'll remove the case 500 since its hard to believe that this is any better than whatever the 500 message is there now. 

However let's also separtely get the 500 message improved.